deepchecks/nlp/utils/text_properties.py

```diff
@@ -35,6 +37,41 @@
 MODELS_STORAGE = pathlib.Path(__file__).absolute().parent / '.nlp-models'
 FASTTEXT_LANG_MODEL = 'https://dl.fbaipublicfiles.com/fasttext/supervised-models/lid.176.bin'
 
+text_blob_cache = {}
+
+
+def _sample_for_property(text: str, mode: str = 'words', limit: int = 10000, random_seed: int = 42):
+    """Get a sample a single text sample for a text property.
+
+    Parameters
+    ----------
+    text : str
+        The text to sample from.
+    mode : str, default 'words'
+        The mode to sample in. Can be either 'characters', 'words' or 'sentences'.
+    limit : int, default 1000
+        The maximum number of words or sentences to sample.
+    """
+    np.random.seed(random_seed)
+    if mode == 'characters':
+        if len(text) <= limit:
+            return text
+        else:
+            all_units = np.random.choice(list(text), size=limit, replace=False)
+            return ''.join(all_units)
+    elif mode == 'words':
+        all_units = re.split(r'\W+', text)
+        if len(all_units) > limit:
+            all_units = np.random.choice(all_units, size=limit, replace=False)
+        return ' '.join(all_units)
+    elif mode == 'sentences':
+        all_units = sent_tokenize(text)
```

Comment:
need to do the "try download" thing like other places where downloadable stuff is used
Is downloadable stuff is used?

Reply:
Technically already done in the property itself. But can move to there.

Reply:
In case this code is later called by other properties

Reply:
Done in the other PR